internal/webhook/external/handlers/refresh_warehouse.go

```diff
@@ -0,0 +1,104 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/api"
+	"github.com/akuity/kargo/internal/indexer"
+	"github.com/akuity/kargo/internal/logging"
+	"github.com/akuity/kargo/internal/webhook/external/providers"
+)
+
+// NewRefreshWarehouseWebhook handles push events for the designated provider.
+// After the provider has been resolved and the request has been authenticated,
+// the kubeclient is queried for all warehouses that contain a subscription
+// to the repo in question. Those warehouses are then patched with a special
+// annotation that signals down stream logic to refresh the warehouse.
+func NewRefreshWarehouseWebhook(
+	p providers.Provider,
+	log *logging.Logger,
+	kClient client.Client,
+) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Info("authenticating request")
+		if err := p.Authenticate(r); err != nil {
+			log.Error(err, "failed to authenticate request")
+			http.Error(w,
+				"failed to authenticate request",
+				http.StatusUnauthorized,
+			)
+			return
+		}
+		log.Info("request authenticated")
```

Comment:
Let's try to keep messages that you don't want to see in production at debug or lower.

```suggestion
		logger.Debug("request authenticated")
```